Simplify and deduplicate test code

mulhod opened the pull request:
Closes #635.

pep8speaks commented:
Hello @mulhod! Thanks for updating this PR. 





* In the file [`tests/test_classification.py`](https://github.com/EducationalTestingService/skll/blob/dfb3937232cabfb602f08d09198cba36c3a0e9a6/tests/test_classification.py):

> [Line 716:101](https://github.com/EducationalTestingService/skll/blob/dfb3937232cabfb602f08d09198cba36c3a0e9a6/tests/test_classification.py#L716): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 100 characters)













* In the file [`tests/test_input.py`](https://github.com/EducationalTestingService/skll/blob/dfb3937232cabfb602f08d09198cba36c3a0e9a6/tests/test_input.py):

> [Line 1141:101](https://github.com/EducationalTestingService/skll/blob/dfb3937232cabfb602f08d09198cba36c3a0e9a6/tests/test_input.py#L1141): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 100 characters)

hlepp commented on tests/test_ablation.py:
```diff
@@ -10,51 +10,42 @@
 
 import csv
 import json
-import os
```
Why remove import?

mulhod replied:
It was only being used for `os.unlink`, which isn't being used anymore. Now `pathlib.Path.unlink` is being used (but it's actually used in a helper method in `tests.utils` so that the input argument can be coerced to a `Path` (i.e. so that we don't have to care about whether the path is a `str` or a `Path`).

mulhod commented:
This PR removes over 450 lines from the tests!

desilinguist reviewed (changes requested):
This is fantastic, @mulhod! Thanks for taking this on – this is a really nice improvement of the codebase. 

I made a couple of minor suggestions. I also have a couple of broader suggestions: 

1. Can you run the tests locally to make sure that all the created files are actually being cleaned up properly? As long as we are working on this, it would be good to make sure of those.

2. I am a little concerned about changing the `from tests.utils import X` statements to `from .utils import X` in terms of the impact this will have on the testing of the PyPi and conda packages. In those scenarios, we install the conda package, do a sparse checkout of the skll tests and then run the tests. I think you should be able to test this by making a local conda package and simulating this scenario.

Thanks again!

mulhod commented on tests/test_classification.py:
```diff
@@ -48,68 +48,39 @@
-        os.unlink(join(output_dir, 'float_class_predictions.tsv'))
-
-    for output_file in glob(join(output_dir, 'train_test_single_file_*')):
-        os.unlink(output_file)
```
Good catch!

mulhod commented on tests/test_commandline_utils.py:
```diff
@@ -60,95 +60,64 @@
-
-    for ffile in glob(join(other_dir, 'features', 'features*')):
-        os.unlink(ffile)
+    for glob_pattern in [join(output_dir, 'test_generate_predictions.tsv'),
```
Good catch again! This actually maintains the original code:
```python
   for f in glob(join(output_dir, 'test_generate_predictions.tsv')):
        os.unlink(f)
```
But it should be fixed since it's not a glob pattern.

mulhod commented:
> This is fantastic, @mulhod! Thanks for taking this on – this is a really nice improvement of the codebase.
> 
> I made a couple of minor suggestions. I also have a couple of broader suggestions:
> 
> 1. Can you run the tests locally to make sure that all the created files are actually being cleaned up properly? As long as we are working on this, it would be good to make sure of those.
> 2. I am a little concerned about changing the `from tests.utils import X` statements to `from .utils import X` in terms of the impact this will have on the testing of the PyPi and conda packages. In those scenarios, we install the conda package, do a sparse checkout of the skll tests and then run the tests. I think you should be able to test this by making a local conda package and simulating this scenario.
> 
> Thanks again!

Thanks, @desilinguist! I decided to revert the parts that used relative imports for now. It's a minor change, anyway. I have run all the tests and fixed a couple cases (one that you pointed out!). I think most of the cases were actually issues with the tests that didn't have anything to do with my changes. But, this gave me the chance to really see what files are left behind, so I just fixed any I saw.

desilinguist commented:
Thanks so much for doing that @mulhod !

desilinguist reviewed (approved):
LGTM!  👍
